server/game/cards/22-BtB/Highgarden.js

```diff
@@ -0,0 +1,140 @@
+        ));
+
+        let numRevealed = revealed.length;
+
+        this.game.promptForSelect(this.controller, {
+            activePromptTitle: `Select up to ${numRevealed} characters`,
+            source: this,
+            numCards: numRevealed,
+            mode: 'upTo',
+            cardCondition: card => card.getType() === 'character' && card.location === 'play area',
+            onSelect: (player, cards) => this.charactersSelect(player, cards, numRevealed),
+            onCancel: (player) => this.charactersCancel(player)
+        });
+        return true;
+    }
+
+    revealCancel(player) {
+        this.game.addAlert('danger', '{0} did not reveal any cards for {1}', player, this);
+
+        return true;
+    }
+
+    charactersSelect(player, cards, numRevealed) {
+        this.strCharacterMap = new Map();
+        this.remainingCards = cards;
+        this.remainingStr = numRevealed * 2;
+        
+        this.game.queueSimpleStep(() => this.calculateNextCharacter(player));
+        
+        this.game.queueSimpleStep(() => {
+            for(let [str, cards] of this.strCharacterMap) {
+                for(let card of cards) {
+                    this.untilEndOfPhase(ability => ({
+                        match: card,
+                        effect: ability.effects.modifyStrength(str)
+                    }));
+                }
+            }
+            // Groups chat messages by STR's given. eg. "give A, B and C +2 STR", or "give A +4 STR and B +2 STR"
+            let strMessages = Array.from(this.strCharacterMap, ([str, cards]) => Message.fragment('{cards} +{str} STR', { cards, str }));
+
+            this.game.addMessage('{0} then uses {1} to give {2} until the end of the phase', player, this, strMessages);
+
+            return true;
+        });
+        return true;
+    }
+
+    calculateNextCharacter(player) {
+        let difference = (this.remainingStr / 2) - this.remainingCards.length;
+        if(this.remainingCards.length === 1 && this.remainingStr > 0) {
+            // If one character remaining, give this character the remaining STR
+            this.addCharacterToStrMap(player, this.remainingStr);
+        } else if(difference === 0) {
+            // If STR can only be spread evenly, give this character +2 STR. This will be hit for remaining characters
+            this.addCharacterToStrMap(player, 2);
+        } else {
+            // Otherwise, ask how much STR to give this character
+            let buttons = range(1, 2 + difference).reverse().map(amount => {
+                let str = amount * 2;
+                return { text: str.toString(), method: 'addCharacterToStrMap', arg: str };
+            });
+            this.game.promptWithMenu(player, this, {
+                activePrompt: {
+                    menuTitle: `Select +STR for ${this.remainingCards[0].name}`,
+                    buttons: buttons
+                },
+                source: this
+            });
```

Comment:
Would love it if the overall prompt was something like clicking a character multiple times to choose them multiple times, instead of choosing characters then button prompts for how much STR. But looking at the existing select card prompt code, this isn't easily doable. Perhaps as a future improvement.

Reply:
I do agree here; maybe a 'repeatable' selector type? Not sure, but it would need to be something that maybe has a counter on each card clicked, to track how many times you've selected it.

For now, this solution seems the most elegant & responsive to me.